includes/compat/subscriptions/class-wc-payments-email-failed-renewal-authentication.php

```diff
@@ -101,7 +101,7 @@ public function get_content_plain() {
 	 * @return string
 	 */
 	public function get_authorization_url( $order ) {
-		return add_query_arg( 'wcpay-confirmation', 1, $order->get_checkout_payment_url( false ) );
+		return add_query_arg( 'wcpay-confirmation', 1, $order->get_checkout_payment_url( false ) ); // nosemgrep: audit.php.wp.security.xss.query-arg -- server generated url is passed in.
```

Comment:
Can we double-check if those server-generated URLs are sanitized and not affected by user input?

Reply:
The url is not affected by user input - [doc](https://woocommerce.github.io/code-reference/classes/WC-Order.html#method_get_checkout_payment_url)